packages/lit-dev-content/site/docs/data/task.md

```diff
@@ -0,0 +1,243 @@
+
+While Lit's reactive update lifecycle is batched and asynchronous, Lit _templates_ always renders synchronously. Handling async data requires collecting that data and triggering a new update, which will render the data synchronously when it runs.
+
+The `@lit-labs/task` package provides a `Task` reactive controller to help manage this and more aspects of async operations.
+
+```ts
+import {Task} from '@lit-labs/task';
+
+class MyElement extends LitElement {
+  @property()
+  productId?: number;
+
+  private _apiTask = new Task(
+    this,
+    {
+      task: async ([productId]) =>
+        const response = await fetch(`example.com/product/${productId}`);
+        return response.json();
+      ),
+      args: () => [this.productId]
+    }
+  );
+
+  render() {
+    return html`
+      ${this._apiTask.render({
+        pending: () => html`
+          <div class="loading">Loading product...</div>
+        `,
+        complete: (product) => html`
+          <h1>${product.name}</h1>
+          <p>${product.price}</p>
+        `,
+      })}
+    `;
+  }
+}
+```
+
+The Task controller takes care of a number of things needed to properly manage async work:
+- Gathers task arguments when the host updates
+- Starts task functions when the arguments change
+- Tracks the task status (initial, pending, complete, or error)
+- Saves the completion value or error of the task function
+- Triggers a host update when the task changes status
+- Handles race conditions, ensuring that only the latest task invocation completes the task
+- Renders the correct template for the current task status
+
+This removes most of the boilerplate for correctly using async data from your code, and ensures robust handling of race conditions and other edge-cases.
+
+## Installation
+
+```bash
+npm i @lit-labs/task
+```
+
+## Usage
+
+`Task` is a [reactive controller](../../composition/controllers/), so it can respond to and trigger updates to Lit's reactive update lifecycle.
+
+You'll generally have one Task object for each logical task that your component needs to perform. Install tasks as fields on your class:
+
+```ts
+class MyElement extends LitElement {
+  private _myTask = new Task(this, {/*...*/});
+}
+```
+
+As a class field, the task status and value are easily available:
+
+```ts
+this._task.value
+```
+
+### Arguments and the task function
+
+The most critical part of a task declaration is the _task function_. This is the function that does the actual work. The Task controller will automatically call the task function with arguments, which have to be supplied with a separate callback. Arguments are separate so they can be checked for changes and the task function only called if the arguments have changed.
+
+The task function takes the task arguments as an _array_ passed as the first parameter
+
+```ts
+new Task(this, {
+  task: async ([arg1, arg2]) => {
+    // do async work here
+  },
+  args: () => [this.field1, this.field2]
+})
+```
```

Comment:
You never show or mention the shorthand of `new Task(this, ()=> {...}, () => [...])`. Should we mention it, or remove it from Task?